packages/app/src/components/swap/swap.tsx

```diff
@@ -57,20 +61,10 @@ const Swap: FC<Props> = ({
     .mul(11)
     .div(10);
 
-  const { state: _swapTx, send: swapExactAmountIn } = useContractFunction(
-    contracts[protocolName].swap,
-    "swapExactAmountIn"
-  );
+  const { account } = useEthers();
 
-  const { state: _approveTokenTx, send: approveUnderlyingTokens } = useContractFunction(
-    contracts[protocolName].token,
-    "approve"
-  );
-
-  const { state: _approveTenderTx, send: approveTenderTokens } = useContractFunction(
-    contracts[protocolName].tenderToken,
-    "approve"
-  );
+  const allowance = useTokenAllowance(tokenSendedAddress, account, addresses[protocolName].swap);
+  const isTokenApproved = allowance != null && allowance.gte(sendTokenAmount === "" ? "0" : sendTokenAmount);
```

Comment:
`!=` is not allowed by the linter and `!==` lead to different behaviour with this code. Please use the suggestion. 

Also need to convert the `sendTokenAmount` to the wei amount. 

```suggestion
  const allowance: BigNumber | undefined = useTokenAllowance(tokenSendedAddress, account, addresses[protocolName].swap);
  const isTokenApproved = allowance?.gte(utils.parseEther(sendTokenAmount || "0")
```

Reply:
it is allowed by the linter, and we do not want to make a difference between `null` and `undefined`. That would be a big source of developer errors.  Also type for `allowance` is inferred, there's no need for explicitly adding it.

see:
![image](https://user-images.githubusercontent.com/2270661/124469966-4a3b4600-dd9b-11eb-9070-0e5e0810ba5c.png)